### ISSUE TITLE
Using external table created from FDW crashes with wrong configuration

### ISSUE THREAD

**Issue by chrishajas** [greenplum-db/gpdb#14528]
## Bug Report
Only tested this on main branch.

Repro:
```
CREATE FOREIGN TABLE foo (                                                                                                                                                                                                                                  a int,                                                                                                                                                                                                                                                        b int                                                                                                                                                                                                                                                     ) SERVER gp_exttable_server                                                                                                                                                                                                                                   OPTIONS (format 'csv', delimiter ',', "null" '', escape '"', quote '"', format_type 'c', location_uris 'file://chajas-a01.vmware.com/tmp/test.csv', log_errors 'f', encoding '6', is_writable 'false');

explain select * from foo;
```

I left out `execute_on 'master'` from this, which causes the crash. 

I'm also a bit confused about `execute_on` vs `mpp_execute` behavior. If I use `execute_on 'all segments'` when creating the FDW we still execute on the coordinator

**Comment by jingwen-yang-yjw** [comment]
> 

Hi @chrishajas ,

now we usually use `create external table` syntax to create an external table in gpdb, which provides more permission check.

If we want to use `create foreign table` syntax to create an external table in gpdb, we must specify `execute_on` option.
This issue is caused by missing a permission check for `execute_on` option when using `create foreign table` syntax to create an external table in gpdb.
I'll add a permission check for `execute_on` option in gp_exttable_fdw validator.


**execute_on vs mpp_execute**

- For historical reasons, `execute_on` option is only for gpdb's external table.

   For `create external table` syntax, we get `execute_on` option from `EXECUTE ... ON ...` clause:
   (Look `create external table` syntax for more information about `execute_on` option.
    https://gpdb.docs.pivotal.io/6-3/ref_guide/sql_commands/CREATE_EXTERNAL_TABLE.html)
   ```
   CREATE EXTERNAL TABLE
   ...
   EXECUTE 'command' [ON ALL 
                          | MASTER
                          | number_of_segments
                          | HOST ['segment_hostname'] 
                          | SEGMENT segment_id ]
  ...
  ```
  
    If we use `create foreign table` syntax to create an external table in gpdb, we must specify `execute_on` option.
    Compared to `mpp_execute` option, `execute_on` option can have more choices.
    And now, we can **NOT** specify `mpp_execute` option for gpdb's external table, it does't work.
    
    

-  `mpp_execute` option is for other foreign table.
   ```
   CREATE FOREIGN TABLE ... OPTIONS ([mpp_execute {master, any, all segments}])
   ```

**Comment by adam8157** [comment]
Fixed by #14797
